core-bundle/src/Resources/contao/dca/tl_templates.php

```diff
@@ -235,11 +237,25 @@ public function addNewTemplate()
 
 		/** @var SplFileInfo[] $files */
 		$files = Contao\System::getContainer()->get('contao.resource_finder')->findIn('templates')->files()->name('/\.html5$/');
+		$projectDir = Contao\System::getContainer()->getParameter('kernel.project_dir');
 
 		foreach ($files as $file)
 		{
-			$strRelpath = Contao\StringUtil::stripRootDir($file->getPathname());
-			$strModule = preg_replace('@^(vendor/([^/]+/[^/]+)/|system/modules/([^/]+)/).*$@', '$2$3', strtr($strRelpath, '\\', '/'));
+			// Do not use "StringUtil::stripRootDir()" here, because for
+			// symlinked bundles, the path will be outside the project dir.
+			$strRelpath = Path::makeRelative($file->getPathname(), $projectDir);
+
+			$modulePatterns = array(
+				"vendor/(?'module'[^/]+/[^/]+)",
+				"\\.\\..*(?'module'[^/]+/[^/]+)/(?:src/Resources/contao/templates|contao/templates)",
+				"system/modules/(?'module'[^/]+)"
+			);
+			preg_match('@^(?:' . implode('|', $modulePatterns) . ')/.*$@UJ', $strRelpath, $matches);
+
+			// Use the matched "module" group and fall back to the full
+			// directory path (e.g. "contao/templates" in the app).
+			$strModule = $matches['module'] ?? dirname($strRelpath);
```

Comment:
```suggestion
			$modulePatterns = array(
				"vendor/([^/]+/[^/]+)",
				"\\.\\..*?([^/]+/[^/]+)/(?:src/Resources/contao/templates|contao/templates)",
				"system/modules/([^/]+)"
			);
			preg_match('@^(?|' . implode('|', $modulePatterns) . ')/.*$@', $strRelpath, $matches);

			// Use the matched "module" group and fall back to the full
			// directory path (e.g. "contao/templates" in the app).
			$strModule = $matches[1] ?? dirname($strRelpath);
```

I’d write the regex without the `UJ` modifiers, but that’s just a nitpick, I’m fine with this version too. ☺️
Duplicate group numbers: <https://www.pcre.org/current/doc/html/pcre2pattern.html#SEC15>